tests/app/main/test_frameworks.py

```diff
@@ -2188,24 +2557,45 @@ def test_contract_review_page_aborts_if_visited_when_information_required_to_ret
             self.login()
             data_api_client.get_framework.return_value = get_g_cloud_8()
             return_supplier_framework.return_value = self.supplier_framework(
+                framework_slug='g-cloud-8',
                 on_framework=True,
                 declaration={
                     "primaryContact": "contact name",
                     "primaryContactEmail": "email@email.com",
                     "nameOfOrganisation": u"£unicodename"
                 },
-                agreement_details={"signerName": "signer_name", "signerRole": "signer_role"},
-                agreement_path=None  # no file has been uploaded
             )['frameworkInterest']
+            data_api_client.get_framework_agreement.return_value = self.framework_agreement(
+                signed_agreement_details={"signerName": "signer_name", "signerRole": "signer_role"}
+                # No file has been uploaded
+            )
 
             # no file has been uploaded
             s3.return_value.get_key.return_value = None
 
             res = self.client.get(
-                "/suppliers/frameworks/g-cloud-8/contract-review"
+                "/suppliers/frameworks/g-cloud-8/234/contract-review"
             )
             assert res.status_code == 404
 
+    @mock.patch('app.main.views.frameworks.check_agreement_is_related_to_supplier_framework_or_abort')
+    @mock.patch('dmutils.s3.S3')
+    def test_we_abort_if_agreement_does_not_match_supplier_framework(
+        self, s3, check_agreement_is_related_to_supplier_framework_or_abort, return_supplier_framework, data_api_client
+    ):
+        self.login()
+        data_api_client.get_framework.return_value = get_g_cloud_8()
+        data_api_client.get_framework_agreement.return_value = self.framework_agreement(supplier_id=2345)
+        supplier_framework = self.supplier_framework(framework_slug='g-cloud-8', on_framework=True)['frameworkInterest']
+        return_supplier_framework.return_value = supplier_framework
+        s3.return_value.get_key.return_value = None
+
+        res = self.client.get("/suppliers/frameworks/g-cloud-8/234/contract-review")
+        check_agreement_is_related_to_supplier_framework_or_abort.assert_called_with(
```

Comment:
There isn't any point in making the mock actually abort and checking the response is 404 because that's just testing that the mocking framework does what it's supposed to, and wouldn't test any of our code.